### PR TITLE
[Bug](shell) fix wrong condition expression in script

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
cc @adonis0147 this condition always return false now

**Comment by adonis0147** [comment]
Would you like to provide some steps to reproduce the issue? I ran `bash -x run-regression-test.sh` on my machine but found the condition passed.

File: run-regression-test.sh [hunk 1]
```diff
@@ -141,7 +141,7 @@ if [[ "${RUN}" -ne 1 ]]; then
     exit 0
 fi
 
-if [[ ! -f ${RUN_JAR} ]]; then
+if [ ! -f ${RUN_JAR} ]; then
```
Comment (adonis0147) [inline run-regression-test.sh:144]: Please change to this to pass the `shellcheck`.

```suggestion
if ! test -f ${RUN_JAR:+${RUN_JAR}}; then
```

**Review by adonis0147 — CHANGES_REQUESTED** [review]

**Comment by adonis0147** [comment]
Finally, I got the issue which the pr aims at resolving.

When we run `run-regression-test.sh`, the script always build the jar for regression test framework even if the jar was built before.

**Review by adonis0147 — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**